### PR TITLE
Fix integration test errors

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
`./test/project.go` didn't run in integration tests.
- Rename to `./tests/project_test.go`
- Fix labels not found in integration tests if update request that doesn't have labels values
- Improve error messages when project not found.
- Add unit and integration test for the project not found.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Throw an error if we can't find the project.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1208

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/validation/project_validator.go [hunk 1]
```diff
@@ -63,10 +63,17 @@ func ValidateProjectAndDomain(
 			"failed to validate that project [%s] and domain [%s] are registered, err: [%+v]",
 			projectID, domainID, err)
 	}
+
+	if project.State == nil {
+		return errors.NewFlyteAdminErrorf(codes.NotFound,
+			"project [%s] is not found", projectID)
+	}
+
```
Comment (pmahindrakar-oss) [inline pkg/manager/impl/validation/project_validator.go:71]: Trying to understand why we need this when the Get() returns an error when no records are found for the project
https://github.com/flyteorg/flyteadmin/blob/master/pkg/repositories/gormimpl/project_repo.go#L45

Comment (katrogan) [reply pkg/manager/impl/validation/project_validator.go:71]: hey @pmahindrakar-oss see this https://github.com/flyteorg/flyteadmin/pull/216 from the linked issue for an example of why we need this

Comment (katrogan) [reply pkg/manager/impl/validation/project_validator.go]: awesome find, thank you @pingsutw

**Review by katrogan — APPROVED** [review]

File: tests/project_test.go [hunk 2]
```diff
@@ -88,9 +103,13 @@ func TestUpdateProjectDescription(t *testing.T) {
...
 	assert.NotEmpty(t, projectsUpdated.Projects)
 
-	// Verify that the project's Name has not been modified but the Description has.
-	updatedProject := projectsUpdated.Projects[0]
-	assert.Equal(t, updatedProject.Id, "potato")                     // unchanged
+	// Verify that the project's ID has not been modified but the Description has.
+	var updatedProject admin.Project
+	for _, project := range projectsUpdated.Projects {
+		if project.Id == "potato" {
+			updatedProject = project
```
Comment (katrogan) [inline tests/project_test.go:110]:
```
tests/project_test.go:110:19: cannot use project (type *admin.Project) as type admin.Project in assignment
tests/project_test.go:172:19: cannot use project (type *admin.Project) as type admin.Project in assignment
```

File: pkg/manager/impl/project_manager.go [hunk 3]
```diff
@@ -119,6 +120,10 @@ func (m *ProjectManager) UpdateProject(ctx context.Context, projectUpdate admin.
 
 	// Transform the provided project into a model and apply to the DB.
 	projectUpdateModel := transformers.CreateProjectModel(&projectUpdate)
+	err = mergo.Merge(&projectUpdateModel, oldProject)
```
Comment (katrogan) [inline pkg/manager/impl/project_manager.go:123]: why is this necessary? shouldn't the gorm update call already handle merging?

**Review by katrogan — COMMENTED** [review]
mind updating the PR title?

File: pkg/repositories/transformers/project.go [hunk 4]
```diff
@@ -14,6 +14,14 @@ type CreateProjectModelInput struct {
...
 	stateInt := int32(project.State)
+	if project.Labels == nil {
+		return models.Project{
+			Identifier:  project.Id,
+			Name:        project.Name,
+			Description: project.Description,
+			State:       &stateInt,
+		}
+	}
 	projectBytes, err := proto.Marshal(project)
```
Comment (pmahindrakar-oss) [inline pkg/repositories/transformers/project.go:25]: Can you explain why this check is needed . Seems even if Labels is not passed the proto.Marshal should generate a byte string which is used as labels on L33

File: pkg/repositories/gormimpl/task_repo.go [hunk 5]
```diff
@@ -53,6 +50,9 @@ func (r *TaskRepo) Get(ctx context.Context, input interfaces.Identifier) (models
 			Version: input.Version,
 		})
 	}
+	if tx.Error != nil {
+		return models.Task{}, r.errorTransformer.ToFlyteAdminError(tx.Error)
```
Comment (pmahindrakar-oss) [inline pkg/repositories/gormimpl/task_repo.go:54]: Would be good to cover these error paths if its not too much effort through unit tests

File: pkg/repositories/gormimpl/task_repo.go [hunk 6]
```diff
@@ -42,9 +42,6 @@ func (r *TaskRepo) Get(ctx context.Context, input interfaces.Identifier) (models
 		},
 	}).Take(&task)
 	timer.Stop()
-	if tx.Error != nil {
-		return models.Task{}, r.errorTransformer.ToFlyteAdminError(tx.Error)
-	}
```
Comment (pmahindrakar-oss) [inline pkg/repositories/gormimpl/task_repo.go:47]: Was gorm error overriding the recordNot found and was that the reason you moved it to L53?

File: pkg/repositories/transformers/project.go (hunk 4 above)
Comment (pingsutw) [reply pkg/repositories/transformers/project.go:25]: When I was running integration tests, I found that we will lose labels if the update project request didn't have labels.
We tried to assign `project.label` a `projectBytes` that didn't contain labels, so gorm will update labels value to nil.

File: pkg/repositories/gormimpl/task_repo.go (hunk 5 above)
Comment (pingsutw) [reply pkg/repositories/gormimpl/task_repo.go:54]: I'm working on it

File: pkg/repositories/gormimpl/task_repo.go (hunk 6 above)
Comment (pingsutw) [reply pkg/repositories/gormimpl/task_repo.go:47]: yes, exactly.

File: pkg/manager/impl/project_manager.go (hunk 3 above)
Comment (pingsutw) [reply pkg/manager/impl/project_manager.go]: you're right. updated it, thanks

File: tests/project_test.go (hunk 2 above)
Comment (pingsutw) [reply tests/project_test.go:111]: Fixed it, thanks.

**Review by pmahindrakar-oss — APPROVED** [review]
